examples/tutorials/gmt-config.py

```diff
@@ -0,0 +1,68 @@
+"""
+Configuring PyGMT defaults
+==========================
+
+Default GMT parameters can be set globally or locally using :class:`pygmt.config`
+"""
+
+import pygmt
+
+########################################################################################
+# Configuring default GMT parameters
+# ----------------------------------
+#
+# The user can override default parameters either temporarily (locally) or permanently
+# (globally) using :meth:`pygmt.config`. The full list of default parameters that can be
+# changed can be at :gmt-docs:`gmt.conf.html`.
+#
+# We demonstrate the usage of :meth:`pygmt.config` by configuring a map plot.
+
+# Start with a basic figure
+fig = pygmt.Figure()
+fig.basemap(region=[115, 119.5, 4, 7.5], projection="M10c", frame=True)
+fig.coast(land="black", water="skyblue")
+
+fig.show()
+
+########################################################################################
+# Globally overriding defaults
+# ----------------------------
+#
+# The `MAP_FRAME_TYPE` parameter specifies the style of map frame to use, of which there
+# are 3 options: `fancy` (default, seen above), `plain`, and `inside`.
+#
+# The `FORMAT_GEO_MAP` parameter controls the format of geographical tick annotations.
+# The default uses degrees and minutes. Here we specify the ticks to be a decimal number
+# of degrees.
+
+pygmt.config(MAP_FRAME_TYPE="plain")
+pygmt.config(FORMAT_GEO_MAP="ddd.xx")
```

Comment:
In GMT 6.1.0, each session can have a global gmt.conf file, and each figure can also have its own gmt.conf (see https://docs.generic-mapping-tools.org/6.1/cookbook/features.html#gmt-modern-mode-hierarchical-levels). 

Now these two lines no longer works as we expected. They actually change the defaults of the figure above. To change the defaults of the current figure, we need to move these two lines after `fig=pygmt.Figure()`.

Reply:
So hierarchically one must do something like:

```bash
# Set global (session) configuration

fig = pygmt.Figure()  # This will change the current 'level' from session to figure
# Set figure configuration

etc
```

That makes sense, I'll fix this up (and finish the two PRs I've opened!)

Reply:
How would one trigger the further levels mentioned in the GMT 6.1.0 documentation (`subplot`, `panel`, `inset`)?

Reply:
Currently, `subplot` and `inset` are not implemented in PyGMT yet, but see #412 and #427.

Reply:
Just opened up an issue for `inset` at #663, we can discuss over there. The `subplot` implementation at #427 is also ready for review :smile:

Reply:
I just read it! I'll have a look at it, thanks.